src/app.tsx

```diff
@@ -45,6 +47,15 @@ export default function App() {
     setInput('')
     setIsLoading(true)
     agent.send(userMessage)
+
+    if (timeout) clearTimeout(timeout)
+    const ping = () => {
+      // @ts-ignore
+      timeout = setTimeout(ping, 1000)
+      agent.send('PING')
```

Comment:
Since an email event had no partykit `connection`, I had to store it in memory. Which meant I needed to keep the DO alive with this ping.